### ISSUE TITLE
Towards `-Wall,-Wpedantic,-Werror`

### ISSUE THREAD

**Issue by Rot127** [aquynh/capstone#2113]
We had now several times that the CI in Capstone is fine with several build warnings, but external projects are not.

For `v6` or at least `v7` we should be able to compile CS without any build warnings.
Especially since everything is C and more strict is always better in this case.

- [x] Builds with `-Wall`
  - [x] `-Wenum` (120 warnings)
  - [x] `-Woverlength` (4 warnings)
  - [x] `-Wshadow` (28 warnings) - https://github.com/capstone-engine/capstone/issues/2170
- [ ] Builds with `-Wpedantic` (1926 warnings) in
  - [ ] `ARMDisassembler.c`
  - [ ] `arm.h`
  - [ ] `ARMInstPrinter.c`
  - [ ] `bpf.h`
  - [ ] `capstone.h`
  - [ ] `Mapping.h`
  - [ ] `MCInst.h`
  - [ ] `mips.h`
  - [ ] `ppc.h`
  - [ ] `PPCInstPrinter.c`
  - [ ] `PPCInstPrinter.h`
  - [ ] `riscv.h`
  - [ ] `sh.h`
  - [ ] `sparc.h`
  - [ ] `systemz.h`
  - [ ] `tricore.h`
  - [ ] `TriCoreInstPrinter.c`
  - [ ] `wasm.h`
  - [ ] `xcore.h`

**Issues**

https://github.com/capstone-engine/capstone/issues/2139

**Comment by XVilka** [comment]
Another possible step would be also adding [`clang-tidy`](https://clang.llvm.org/extra/clang-tidy/#clang-tidy) into the CI, see this for example: https://gitlab.com/OpenMW/openmw/-/merge_requests/900/diffs

**Comment by Rot127** [comment]
Yes. Its tracked here: https://github.com/capstone-engine/capstone/issues/2094

**Comment by peace-maker** [comment]
The same can be done for msvc to get more digital eyes on the code. "Werror" can be enabled as well using "/WX". There aren't that many problems when building with `-Wall` https://github.com/capstone-engine/capstone/commit/2d8dad66f834ecf6699e459ed7ef92d059db6b52 so maybe it's already possible for v6.

**Comment by Rot127** [comment]
Agree. There are some warnings disabled though in code with `pragma`. Need to remove those as well at some point.

**Comment by Rot127** [comment]
Made a build with `-Wall` and `-Wpedantic`. We are not so far from passing `-Wall`! And `Wpedantic` seems mostly a problem with our header files. Added a list above.

**Comment by XVilka** [comment]
@imbillow could you please check the Tricore warnings when you have time?

**Comment by add-support-sepolia** [comment]
👍
